_posts/2020-12-20-azure-event-grid.md

```diff
@@ -0,0 +1,55 @@
+
+Ora che abbiamo capito in che scenario ha senso pensare ad Event Grid, cerchiamo di capire un po' meglio cos'e': prima del suo avvento, tutti i servizi di Azure erano una serie di puntini, che potevano cooperare tra di loro, ma con modalita' completamente differenti tra di loro. Event Grid e' la lingua franca con cui mettere in comunicazione tutti questi servizi. La linea che permette di unire tutti i puntini.
+
+I [componenti principali](https://docs.microsoft.com/en-us/azure/event-grid/concepts?WT.mc_id=AZ-MVP-5001625) che caratterizzano Event Grid sono:
+- un evento, ovvero l'informazione che si vuol far "viaggiare" nel sistema
+- una sorgente (source), ovvero il componente che genera l'evento
+- un topic, ovvero l'endpoint a cui viene "inviato" l'evento
+- un handler, ovvero un componente interessato ad un particolare evento
+- una subscription, ovvero il collegamento tra un topic ed un componente interessato all'evento generato
+
+Nell'ecosistema di Azure, una _source_ puo' essere uno dei servizi esposti dalla piattaforma, come per esempio [Azure Container Registry](https://docs.microsoft.com/en-us/azure/event-grid/event-schema-container-registry?WT.mc_id=AZ-MVP-5001625) o [Azure App Configuration](https://docs.microsoft.com/en-us/azure/event-grid/event-schema-app-configuration?WT.mc_id=AZ-MVP-5001625). Un handler puo' essere per esempio una [Azure Function](https://docs.microsoft.com/en-us/azure/event-grid/handler-functions?WT.mc_id=AZ-MVP-5001625) o un [WebHook](https://docs.microsoft.com/en-us/azure/event-grid/handler-webhooks?WT.mc_id=AZ-MVP-5001625).  
+
+Visto che solitamente un'immagine vale piu' di mille parole, ecco tutti i servizi che e' possibile "collegare" sfruttando Event Grid:
+
+![functional model](https://docs.microsoft.com/en-us/azure/event-grid/media/overview/functional-model.png?WT.mc_id=AZ-MVP-5001625)
+
+Il fatto che uno dei possibili handler supportati da Event Grid sia il meccanismo dei [webhook](https://docs.microsoft.com/en-us/azure/event-grid/handler-webhooks?WT.mc_id=AZ-MVP-5001625) apre le porte all'integrazione di sistemi terzi (come per esempio nostre applicazioni) con le sorgenti supportate da Event Grid.
+
+Event Grid mette a disposizione la possibilita' di integrarsi anche dall'altro lato della barricata, ovvero dalla parte di chi produce gli eventi.  
+Tutte le sorgenti _native_ della piattaforma sfruttano dei [topic di sistema](https://docs.microsoft.com/en-us/azure/event-grid/system-topics?WT.mc_id=AZ-MVP-5001625), _system topic_ appunto, ma e' possibile definire dei [topic custom](https://docs.microsoft.com/en-us/azure/event-grid/custom-topics?WT.mc_id=AZ-MVP-5001625), in cui verranno veicolati eventi prodotti da sorgenti esterne ad Azure (come per esempio nostre applicazioni).
+
+Con Azure Event Grid termina l'era del dispendioso "polling" per valutare eventuali cambi di stato del nostro sistema. Ora, qualsiasi scenario _message driven_ e' veramente a distanza di...un messaggio.
```

Comment:
Non mi è chiaro però quale sia la differenza rispetto a qualunque altra tecnologia che "trasporta messaggi" (diciamo JMS, RabbitMq, WMQ, Kafka, solo per limitarmi a quelle che, anche solo superficialmente, conosco). Mi aspettavo un qualche automatismo che "trasforma" in messaggi "cose che già succedono": se ho ben inteso, i _topic di sistema_ svolgono un ruolo simile, approfondirei la cosa descrivendo un paio di scenari in cui "quello che c'è già" inizia ad essere "visto" come _source_ di eventi...

Reply:
Ok, allora aggiungo uno use case. 
L'idea e' proprio quella che, oltre ad essere un "broker" come quelli da te citati, e' l'integrazione di tutta i servizi della piattaforma il vero selling point: non devi piu' essere tu a scrivere codice per "creare" un messaggio quando un file viene caricato in un blob storage, una nuova VM viene creata o una configurazione cambia in App Configuration (ecc), "basta" dire che sei interessato a quel particolare evento ed "Event GRid te lo consegna"

Reply:
Ok, ottimo: allora sì, direi che descrivere qualche use-case può essere d'aiuto.